Feat/uat

Anurag-Wednesday opened the pull request:
### Ticket Link

---

### Related Links

---

### Description

-----Added auto deploy to S3 for Uat on PR functionality
--added workflow for the same

Link to the deployed app

http://uat-tracks.s3-website.ap-south-1.amazonaws.com/feat/UAT/

### Steps to Reproduce / Test

---

---

### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's

---
![2022-02-14 16 38 48](https://user-images.githubusercontent.com/98798828/153853458-38cdf62f-1b3d-4656-90d6-cc1901a8e7f1.gif)

Anurag-Wednesday committed:
added button to redirect to '/tracks'

Anurag-Wednesday committed:
added continuous deployment on PR

Anurag-Wednesday committed:
updated config for history tests

Anurag-Wednesday committed:
updated the workflow

Anurag-Wednesday committed:
updated the workflow to work on merge to master

Anurag-Wednesday committed:
fixed invalid value for 'on' in workflow

alichherawalla commented on app/containers/HomeContainer/index.js:
```diff
@@ -143,6 +143,9 @@ export function HomeContainer({
         <StyledT onClick={handleStoriesClick} data-testid="redirect" id="stories" />
       </RightContent>
+      <button data-testid="redirectToTracks" onClick={() => history.push('/tracks')}>
+        Go to Tracks API
```
react-intl

alichherawalla commented on app/containers/HomeContainer/tests/index.test.js:
```diff
@@ -153,6 +153,9 @@ describe('<HomeContainer /> tests', () => {
     fireEvent.click(getByTestId('redirect'));
     await timeout(500);
     expect(historySpy).toHaveBeenCalledWith('/stories');
+    fireEvent.click(getByTestId('redirectToTracks'));
```
data-testids are usually kebab case

alichherawalla commented on jest.config.json:
```diff
@@ -11,6 +11,7 @@
     "!**/loadable.js",
     "!app/*/*/*/Loadable.js",
     "!**/apiUtils.js",
+    "!**/history.js",
```
this isn't right. We should be adding tests for history.js

Anurag-Wednesday committed:
interim

Anurag-Wednesday committed:
added tests to history and added Tag component

Anurag-Wednesday committed:
added tests to history and added Tag component

Anurag-Wednesday committed:
Added antD components

1. added Notification, Modal, Popover, message Antd components to the Project

Anurag-Wednesday committed:
fixed issue with baseURL

Anurag-Wednesday committed:
fixed issues with baseURL

Anurag-Wednesday committed:
fixed issues with baseURL

Anurag-Wednesday committed:
fixed baseURL

Anurag-Wednesday committed:
updated the baswURL and style for Link

Anurag-Wednesday committed:
added option to upld details using antd components

Anurag-Wednesday committed:
added option to upld tracks using antd components

Anurag-Wednesday committed:
used translated strings

Anurag-Wednesday committed:
Merge branch 'feat/UAT' into feat/CreateTrack

Anurag-Wednesday committed:
fixed incorrect id

alichherawalla commented on app/containers/ItunesProvider/SearchContainer/index.js:
```diff
@@ -103,6 +107,7 @@ export function SearchContainer({
     if (!isEmpty(sTerm)) {
       dispatchGetTrackList(sTerm);
     } else {
+      message.warning('Cleared Search');
```
react-intl please

alichherawalla commented on app/containers/ItunesProvider/SearchContainer/index.js:
```diff
@@ -75,6 +75,10 @@ export function SearchContainer({
   const handleOnActionClick = (track, type) => {
     if (type === 'play') {
+      notification.open({
+        message: 'Now Playing',
```
react-intl

alichherawalla commented on app/components/TrackDetailsComponent/index.js:
```diff
@@ -75,22 +81,37 @@ export function TrackDetailsComponent({
       <Details>
         <TrackTitle>
           <If condition={!isEmpty(trackName)} otherwise={<T data-testid="no-track-name" id="no_track_name" />}>
             <T type="heading" id="track_details_name" data-testid="track_title" values={{ Track: trackName }} />
```
variable names should always be camelCase. Please rename `Track` to `track`

alichherawalla commented on app/components/TrackDetailsComponent/index.js:
```diff
@@ -75,22 +81,37 @@ export function TrackDetailsComponent({
-          <TrackImage data-testid="carousel-image" alt={trackName} src={artworkUrl100} />
-        </StyledCarousel>
+        <Popover
+          title="Title"
```
title-> react-intl

alichherawalla commented on app/components/TrackDetailsComponent/index.js:
```diff
@@ -75,22 +81,37 @@ export function TrackDetailsComponent({
+        <Popover
+          title="Title"
+          content={
+            <T type="subheading" id="track_details_name" data-testid="track_title" values={{ Track: trackName }} />
```
track here should be in camelCase too

Anurag-Wednesday commented on app/components/TrackComponent/index.js:
```diff
@@ -46,41 +46,59 @@ const Title = styled(T)`
+    <>
+      <TrackCard
+        onClick={() => setVisible(true)}
+        data-testid="TrackCard"
```
snakecase

Anurag-Wednesday committed:
Made changes as per the comments

1. Used React-Intl wherever it was missed
2. Split the content for steps in new components
3. fixed casing issues with Id's
4. added tests to ensure complete coverage

Anurag-Wednesday committed:
fixed imports and used ids to translate step title

Anurag-Wednesday committed:
fixed issue for T import

Anurag-Wednesday committed:
updated import statements

Anurag-Wednesday committed:
Merge pull request #10 from Anurag-Wednesday/feat/CreateTrack

Feat/create track

Anurag-Wednesday committed:
updated import for trackCreationSteps
